src/Databases/PostgreSQL/fetchPostgreSQLTableStructure.cpp

```diff
@@ -196,29 +204,59 @@ PostgreSQLTableStructure::ColumnsInfoPtr readNamesAndTypesList(
             }
             else
             {
-                std::tuple<std::string, std::string, std::string, uint16_t, std::string, std::string, std::string> row;
-                while (stream >> row)
-                {
-                    const auto column_name = std::get<0>(row);
-                    const auto data_type = convertPostgreSQLDataType(
-                        std::get<1>(row), recheck_array,
-                        use_nulls && (std::get<2>(row) == /* not nullable */"f"),
-                        std::get<3>(row));
-
-                    columns.push_back(NameAndTypePair(column_name, data_type));
-                    auto attgenerated = std::get<6>(row);
-
-                    attributes.emplace(
-                        column_name,
-                        PostgreSQLTableStructure::PGAttribute{
-                            .atttypid = parse<int>(std::get<4>(row)),
-                            .atttypmod = parse<int>(std::get<5>(row)),
-                            .atthasdef = false,
-                            .attgenerated = attgenerated.empty() ? char{} : char(attgenerated[0]),
-                            .attr_def = {}
-                    });
+                int version = getVersion(tx);
```

Comment:
this is redundant as the version is already fetched in fetchPostgreSQLTableStructure, so you can just pass it through as function argument